src/Location.ml

```diff
@@ -0,0 +1,62 @@
+open Base
+open Lexing
+
+type t =
+  | NoSource
+  | Source of {
+      filename : string; [@equal.ignore]
+      start_line : int; [@equal.ignore]
+      start_column : int; [@equal.ignore]
+      end_line : int; [@equal.ignore]
+      end_column : int; [@equal.ignore]
+    }
+[@@deriving equal, sexp]
+
+type 'a located = { data : 'a; loc : t } [@@deriving equal, sexp]
```

Comment:
Why not just ignore equality for `loc` here?

Reply:
Yes, I did